alpha/lib/model/BatchJob.php

```diff
@@ -390,12 +390,14 @@ public function generateFilePathArr( $sub_type, $version = null, $externalPath =
 		if($externalPath)
 		{
 			$path = '/batchfiles/';
+			$path .= myContentStorage::getScatteredPathFromIntId($this->getPartnerId()) . '/' . $this->generateFileName($sub_type, $version);
```

Comment:
why not use the jobId? 
It will give you a much better distribution of the files.